src/hdmf_zarr/backend.py

```diff
@@ -783,29 +818,31 @@ def write_dataset(self, **kwargs):  # noqa: C901
                 for j, item in enumerate(data):
                     new_item = list(item)
                     for i in refs:
-                        new_item[i] = self.__get_ref(item[i])
+                        new_item[i] = self.__get_ref(item[i], export_source=export_source)
                     dset[j] = new_item
             else:
                 # write a compound datatype
                 dset = self.__list_fill__(parent, name, data, options)
         # Write a dataset of references
         elif self.__is_ref(options['dtype']):
-            if isinstance(data, RegionBuilder):
-                shape = (1,)
-                type_str = 'region'
-                refs = self.__get_ref(data.builder, data.region)
-            elif isinstance(data, ReferenceBuilder):
+            # TODO: Region References are not yet supported
+            # if isinstance(data, RegionBuilder):
+            #     shape = (1,)
+            #     type_str = 'region'
+            #     refs = self.__get_ref(data.builder, data.region)
```

Comment:
```suggestion
```

The first part of the `if` statement appears to refer to `object` references not `region` references